tuxedo-core/src/types.rs

```diff
@@ -155,10 +160,39 @@ pub struct Output<V> {
 #[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
 #[derive(Encode, Decode, Debug, PartialEq, Eq, Clone, TypeInfo)]
 pub struct PreliminarilyValidTransaction<A> {
-    provides: Vec<Vec<u8>>,
-    requires: Vec<Vec<u8>>,
-    priority: TransactionPriority,
-    intermediate_accumulator_value: Option<A>,
+    pub provides: Vec<Vec<u8>>,
+    pub requires: Vec<Vec<u8>>,
+    pub priority: TransactionPriority,
+    pub intermediate_accumulator_value: Option<A>,
+}
+
+impl<A> Into<ValidTransaction> for PreliminarilyValidTransaction<A> {
+    fn into(self) -> ValidTransaction {
+        ValidTransaction {
+            provides: self.provides.clone(),
+            requires: self.requires.clone(),
+            priority: self.priority,
+            longevity: TransactionLongevity::max_value(),
+            propagate: true,
+        }
+    }
+}
+
+impl<A> From<ValidTransaction> for PreliminarilyValidTransaction<A> {
+    fn from(transaction: ValidTransaction) -> Self {
+        Self {
+            provides: transaction.provides,
+            requires: transaction.requires,
+            priority: transaction.priority,
+            intermediate_accumulator_value: None,
+        }
+    }
+}
```

Comment:
@muraca What is the point of this one? When would we need to go in this reverse direction from valid to preliminarily valid?

Reply:
I'll have to double check tomorrow, but I think I did this because it wasn't possible to use some of the ValidTransactionBuilder's fields